VehicleUI/src/main/java/com/drivequant/drivekit/vehicle/ui/vehicledetail/common/VehicleCustomImageHelper.kt

```diff
@@ -0,0 +1,182 @@
+            intent.putExtra(MediaStore.EXTRA_OUTPUT, uri)
+            val filePath = "file://" + file.absolutePath
+            cameraCallback.onFilePathReady(filePath)
+            startActivityForResult(activity, intent, REQUEST_CAMERA, null)
+        } catch (e: Exception) {
+            DriveKitLog.e(DriveKitVehicleUI.TAG, "Could not open camera: $e")
+        }
+    }
+
+    @Throws(IOException::class)
+    private fun createImageFile(activity: Activity, filename: String): File {
+        // taken from https://developer.android.com/training/camera/photobasics
+        val storageDir = activity.getExternalFilesDir(Environment.DIRECTORY_PICTURES)
+        return File.createTempFile(filename, ".jpg", storageDir)
+    }
+
+    private fun buildMatrixOrientation(context: Context, uri: Uri): Matrix? =
+        getImageOrientation(context, uri)?.let {
+            val matrix = Matrix()
+            matrix.setRotate(it.toFloat())
+            return matrix
+        } ?: run {
+            null
+        }
+
+    private fun getImageOrientation(context: Context, uri: Uri): Int? {
+        var stream: InputStream? = null
+        try {
+            stream = context.contentResolver.openInputStream(uri)
+            val exifInterface = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && stream != null) {
+                ExifInterface(stream)
+            } else {
+                null // DriveKit will no longer support Android 6.0 and 7.0 in Q4 2024.
+            }
+
+            exifInterface?.let {
+                val orientation = it.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL)
+                return when (orientation) {
+                    ExifInterface.ORIENTATION_ROTATE_270 -> 270
+                    ExifInterface.ORIENTATION_ROTATE_180 -> 180
+                    ExifInterface.ORIENTATION_ROTATE_90 -> 90
+                    else -> null
+                }
+            }
+        } catch (e: Exception) {
+            DriveKitLog.e(DriveKitVehicleUI.TAG, "Couldn't get image orientation: $e")
+        } finally {
+            stream?.close()
+        }
+        return null
+    }
+
+    private fun Pair<Int, Int>.computeNewImageSize(context: Context): Pair<Int, Int> {
+        val originalWidth = this.first
+        val originalHeight = this.second
+
+        val deviceScreenWidth = (context.resources.displayMetrics.widthPixels * context.resources.displayMetrics.density).toInt()
+        val imageViewHeight = context.resources.getDimension(R.dimen.dk_vehicle_image_height).convertDpToPx()
+
+        if (originalWidth > deviceScreenWidth && originalHeight > imageViewHeight) {
+            val ratio = originalWidth.toFloat() / deviceScreenWidth.toFloat()
+            val newHeight = (originalHeight / ratio).toInt()
+            return Pair(deviceScreenWidth, newHeight)
+        } else {
+            return Pair(originalWidth, originalHeight)
```

Comment:
Pas besoin de recréer un objet `Pair` je pense.
```suggestion
            return this
```